packages/test/src/workflows/signal-handlers-clear.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Workflow used by integration-tests.ts: signal handlers can be cleared
+ *
+ * @module
+ */
+
+import { setHandler, sleep } from '@temporalio/workflow';
+import { unblockSignal } from './definitions';
+
+export async function signalHandlersCanBeCleared(): Promise<number> {
+  let counter = 0;
+
+  // Give time for signals to be received and buffered BEFORE the handler is configured. See issue #612 for details.
+  await sleep('20ms');
```

Comment:
Might be better to switch this to a unit test instead (see test-workflows.ts), I can help with that if you want.

Reply:
Fair enough, I had also considered this option.

That doesn't look too difficult. I'll ping you if I need some help on this.